src/main/java/backend/Logic.java

```diff
@@ -206,14 +201,41 @@ public Model getRepo(String repoId) {
      * @return True if label replacement on GitHub was a success, false otherwise.
      */
     public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> newLabels) {
+        boolean isLabelsChanged = updateIssueLabels(issue, newLabels);
+
+        if (isLabelsChanged) {
+            logger.info("Changing labels for " + issue + " on UI");
+            refreshUI();
+        }
+
+        return updateIssueLabelsOnRepo(issue, newLabels);
+    }
```

Comment:
If you follow the suggested code,
1. `isLabelsChanged(issue, newLabels)` should be tested for the correctness of the boolean condition. e.g. new labels is a subset of originals
2. this method has only two paths, and needs two test cases only. i.e. labels changed case and labels unchanged case.